contracts/src/v0.8/dev/automation/upkeeps/ProxyBalanceMonitor.sol

```diff
@@ -0,0 +1,287 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.4;
+
+import "../../../ConfirmedOwner.sol";
+import "../../../interfaces/automation/KeeperCompatibleInterface.sol";
+import "../../../vendor/openzeppelin-solidity/v4.7.0/contracts/security/Pausable.sol";
+import "../../../vendor/openzeppelin-solidity/v4.7.0/contracts/token/ERC20/IERC20.sol";
+import "../../../vendor/openzeppelin-solidity/v4.7.3/contracts/utils/structs/EnumerableSet.sol";
+
+interface IAggregatorProxy {
+  function aggregator() external view returns (address);
+}
+
+interface IAggregator {
+  function linkAvailableForPayment() external view returns (int256 availableBalance);
```

Comment:
interesting, this is not `uint256 availableBalance` ! I wonder if we handle negatives properly

Reply:
Yup OCR contracts could potentially have a negative balance (this happens in VRF and CCIP).